src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/JsonContext/HighLowTemps.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
+
+namespace System.Text.Json.Tests.Serialization
+{
+    internal partial class JsonContext : JsonSerializerContext
+    {
+        private JsonTypeInfo<HighLowTemps> _HighLowTemps;
+        public JsonTypeInfo<HighLowTemps> HighLowTemps
+        {
+            get
+            {
+                if (_HighLowTemps == null)
+                {
+                    JsonConverter customConverter;
+                    if (Options.Converters.Count > 0 && (customConverter = GetRuntimeProvidedCustomConverter(typeof(HighLowTemps))) != null)
+                    {
+                        _HighLowTemps = JsonMetadataServices.CreateValueInfo<HighLowTemps>(Options, customConverter);
+                    }
+                    else
+                    {
+                        JsonTypeInfo<HighLowTemps> objectInfo = JsonMetadataServices.CreateObjectInfo<HighLowTemps>();
+                        _HighLowTemps = objectInfo;
+    
+                        JsonMetadataServices.InitializeObjectInfo(
+                            objectInfo,
+                            Options,
+                            createObjectFunc: static () => new HighLowTemps(),
+                            HighLowTempsPropInitFunc,
+                            default);
+                    }
+                }
+
+                return _HighLowTemps;
+            }
+        }
+
+        private static JsonPropertyInfo[] HighLowTempsPropInitFunc(JsonSerializerContext context)
+        {
+            JsonContext jsonContext = (JsonContext)context;
+            JsonSerializerOptions options = context.Options;
+
+            JsonPropertyInfo[] properties = new JsonPropertyInfo[2];
+
+            properties[0] = JsonMetadataServices.CreatePropertyInfo<int>(
+                options,
+                isProperty: true,
+                declaringType: typeof(HighLowTemps),
+                propertyTypeInfo: jsonContext.Int32,
+                converter: null,
+                getter: static (obj) => { return ((HighLowTemps)obj).High; },
+                setter: static (obj, value) => { ((HighLowTemps)obj).High = value; },
+                ignoreCondition: default,
+                numberHandling: default,
+                propertyName: "High",
```

Comment:
nit (for future): using `nameof(HighLowTemps.High)` is a bit less error-prone and will give a nice compiler error if properties are removed\renamed (but so will the code gen'd for the getter and setter, so not all that important).

Reply:
makes sense to update it. I've done so in this PR, but will update it in the generator in a follow up.